src/models/modelTests/MyAccount.test.ts

```diff
@@ -54,34 +67,184 @@ describe("MyAccountModel", () => {
+          bibId: "22002760",
+          isNyplOwned: true,
+          catalogHref: "https://borrow.nypl.org/search/card?recordId=22002760",
+        },
+        {
+          bibId: "21317166",
+          canFreeze: false,
+          catalogHref: "https://borrow.nypl.org/search/card?recordId=21317166",
+          frozen: false,
+          id: "42273371",
+          isNyplOwned: true,
+          isResearch: false,
+          patron: "2772226",
+          pickupByDate: null,
+          pickupLocation: {
+            code: "mp",
+            name: "Morris Park",
+          },
+          status: "REQUEST PENDING",
+          title: "2017 Tony Award Season.",
+        },
+      ])
+    })
+    it("can return fines", async () => {
+      const mockSierraClient = {
+        get: async (path) => Promise.resolve(fines),
+      }
+      const fetcher = new MyAccount(mockSierraClient, "12345")
+      const processedFines = await fetcher.getFines()
+      expect(processedFines).toStrictEqual({
+        total: 14.99,
+        entries: [
+          {
+            detail: "Replacement",
+            amount: 14.99,
+            date: "June 15, 2023",
+          },
+        ],
+      })
+    })
+  })
+  describe("getResearchAndOwnership", () => {
+    it("can handle no varfields", () => {
+      expect(MyAccount.getResearchAndOwnership({})).toStrictEqual({
+        isResearch: false,
+        isNyplOwned: true,
+      })
+    })
+    it("can handle missing 910 field", () => {
+      expect(
+        MyAccount.getResearchAndOwnership({
+          varFields: [{ marcTag: "666" }],
+        })
+      ).toStrictEqual({
+        isResearch: true,
+        isNyplOwned: false,
+      })
+    })
+    it("can handle a otf record", () => {
+      expect(
+        MyAccount.getResearchAndOwnership({
+          varFields: [
+            { marcTag: "910", subfields: [{ tag: "a", content: "RLOTF" }] },
+          ],
+        })
+      ).toStrictEqual({
+        isResearch: true,
+        isNyplOwned: false,
+      })
+    })
+    it("can handle an nypl research record", () => {
+      expect(
+        MyAccount.getResearchAndOwnership({
+          varFields: [
+            { marcTag: "910", subfields: [{ tag: "a", content: "RL" }] },
+          ],
+        })
+      ).toStrictEqual({
+        isResearch: true,
+        isNyplOwned: true,
+      })
+    })
+  })
 
-  describe("building model", () => {
+  describe("MyAccountFactory", () => {
     it("builds Account data model", async () => {
-      MyAccount.fetchCheckouts = async () => checkouts
-      MyAccount.fetchHolds = async () => holds
-      MyAccount.fetchPatron = async () => patron
-      MyAccount.fetchFines = async () => fines
-      MyAccount.fetchBibData = async (entries, recordType) => {
-        if (recordType === "item") {
-          return checkoutBibs
-        } else return holdBibs
-      }
-      const account = await MyAccountFactory("12345")
+      MyAccount.prototype.fetchCheckouts = async () =>
```

Comment:
Does mocking the functions directly, without using the prototype, work? Just curious because the functions are assigned within the class and not as prototype functions but you can still mock it this way.

Reply:
I generally mock instance methods with .prototype. when i remove it, i also get type errors.